app/src/main/java/com/handstandsam/shoppingapp/repository/CheckoutCart.kt

```diff
@@ -1,21 +1,37 @@
 package com.handstandsam.shoppingapp.repository
 
 import com.handstandsam.shoppingapp.models.Item
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.launch
 
-data class CheckoutCart(val itemsInCart: MutableList<Item> = mutableListOf()) {
+data class CheckoutCart(val itemsInCart: MutableList<Item> = mutableListOf()) :
+    CoroutineScope by CoroutineScope(Dispatchers.Default) {
+
+
+    private val channel = ConflatedBroadcastChannel<List<Item>>(itemsInCart.toList())
 
     fun empty() {
         itemsInCart.clear()
```

Comment:
When you clear the cart or remove an item the UI doesn't clear because you aren't pushing the latest value to the channel, right?

Reply:
@pakoito correct. My main goal was to just get the red icon working 😂. You are correct though. This code will only work when items are added, but that is easy to update at some point.

Reply:
Maybe if the channel caches the state the channel could be your state and you won't need the mutable list. That's what we used to do with BehaviorSubject/Relay in Rx!

Reply:
@pakoito - That's a really neat idea!  That would work :-)  I am prepping to back this cart by a DB for my upcoming talk, and not sure exactly if that's my best approach at the moment, but I added in channel updates for clear and remove for you: https://github.com/handstandsam/ShoppingApp/commit/c199c83c72d00e230d8661d52356f983e0af52a1